src/System.Management/src/System/Management/ManagementPath.cs

```diff
@@ -321,8 +322,9 @@ private static string GetWbemPath(IWbemPath wbemPath)
 
                     if (status >= 0 && 0 < bufLen)
                     {
-                        pathStr = new string ('0', (int) bufLen-1);
-                        status = wbemPath.GetText_(flags, ref bufLen, pathStr);
+                        char[] pathChars = new char[(int)bufLen];
+                        status = wbemPath.GetText_(flags, ref bufLen, pathChars);
+                        pathStr = new string(pathChars, 0, Array.IndexOf(pathChars, '\0'));
```

Comment:
This is functionally the same as what was being done previously with StringBuilder, in that StringBuilder marshaling will search for the '\0' and use that when copying back the data.  We could make this more efficient presumably by using `bufLen-1` instead if `status` indicates success, but I'm guessing we're not particularly concerned with performance in this code, and our testing isn't great around this stuff, so it's probably fine as-is.